flybywiresim/aircraft
# feat(fms): run predictions without v-speeds

Fixes #8228 

## Summary of Changes

Computes vertical predictions without a V2 speed entered. Previously, this could not be done because the V2 speed was needed for takeoff path predictions. However, the real plane does do this somehow. I have implemented a quick lookup table to estimate a V2 speed based on the entered weight.

Also
- fixes an issue where predictions would be nonsensical without a flap entry
- fixes the starting point of the vertical profile to line up with the lateral path

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): BlueberryKing

## Testing instructions

- Enter a flight plan
- Make sure there are no predictions on the flight plan page
- Enter a cruise altitude, the ZFW and block fuel
- Make sure predictions are computed on the flight plan page
- Enter V-speeds and a flap setting on the PERF page, make sure nothing breaks.


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
